client/my-sites/customer-home/cards/notices/celebrate-notice/index.jsx

```diff
@@ -47,7 +48,7 @@ const CelebrateNotice = ( {
 
 	const showNextTask = () => {
 		setIsLoading( true );
-		dispatch( skipViewHomeLayout( siteId, currentView ) );
+		skipHomeView( siteId, currentView );
```

Comment:
`useCurrentView` is only ever used to get the `skipHomeView`'s second parameter. Then why not integrate it directly into that hook?

Also, the current code is fragile. What if `showNextTask` is called before the `useHomeLayoutQuery` finished fetching? Then `currentView` will be `undefined`.

Reply:
I've moved the lookup for the current view so it happens within the `useSkipCurrentViewMutation()` hook.

> Also, the current code is fragile. What if `showNextTask` is called before the `useHomeLayoutQuery` finished fetching? Then `currentView` will be `undefined`.

My Home doesn't render _anything_ until `useHomeLayoutQuery` has finished, which means this button (and the ability to call `showNextTask`) won't appear until the current view is a valid value. But to be defensive I've protected the skip call like this:
```typescript
if ( siteId && viewName ) {
	mutate( { siteId, viewName, reminder } );
}
```

Does that address the fragility?